src/test/java/com/conveyal/osmlib/TestUtils.java

```diff
@@ -0,0 +1,121 @@
+        // first, terminate all other user sessions
+        executeAndClose("SELECT pg_terminate_backend(pg_stat_activity.pid) " +
+            "FROM pg_stat_activity " +
+            "WHERE pg_stat_activity.datname = '" + dbName + "' " +
+            "AND pid <> pg_backend_pid()");
+        // drop the db
+        executeAndClose("DROP DATABASE " + dbName);
+    }
+
+    /**
+     * Boilerplate for opening a connection, executing a statement and closing connection.
+     *
+     * @param statement
+     * @return true if everything worked.
+     */
+    private static boolean executeAndClose(String statement) {
+        Connection connection;
+        try {
+            connection = DriverManager.getConnection(pgUrl);
+        } catch (SQLException e) {
+            e.printStackTrace();
+            LOG.error("Error creating new database!");
+            return false;
+        }
+
+        try {
+            connection.prepareStatement(statement).execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
+            LOG.error("Error creating new database!");
+            return false;
+        }
+
+        try {
+            connection.close();
+            return true;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            LOG.error("Error closing connection!");
+            return false;
+        }
+    }
+
+    /**
+     * Generate a new database for isolating a test.
+     *
+     * @return The name of the name database, or null if creation unsucessful
+     */
+    public static String generateNewDB() {
+        String newDBName = randomIdString();
+        if (executeAndClose("CREATE DATABASE " + newDBName)) {
+            return newDBName;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Helper to return the relative path to a test resource file
+     *
+     * @param fileName
+     * @return
+     */
+    public static String getResourceFileName(String fileName) {
+        return "./src/test/resources/" + fileName;
+    }
+
+    /**
+     * Generate a random unique prefix of n lowercase letters.
+     * We can't count on sql table or schema names being case sensitive or tolerating (leading) digits.
+     * For n=10, number of possibilities is 26^10 or 1.4E14.
+     *
+     * The approximate probability of a hash collision is k^2/2H where H is the number of possible hash values and
+     * k is the number of items hashed.
+     *
+     * SHA1 is 160 bits, MD5 is 128 bits, and UUIDs are 128 bits with only 122 actually random.
+     * To reach the uniqueness of a UUID you need math.log(2**122, 26) or about 26 letters.
+     * An MD5 can be represented as 32 hex digits so we don't save much length, but we do make it entirely alphabetical.
+     * log base 2 of 26 is about 4.7, so each character represents about 4.7 bits of randomness.
+     *
+     * The question remains of whether we need globally unique IDs or just application-unique IDs. The downside of
```

Comment:
This code appears to be copied out of gtfs-lib. Since these are separate libraries I guess we have no choice but to duplicate the code. But the javadoc comments contain references to "feeds" and "older databases" which only make sense in the gtfs-lib context.

I'm also wondering why we need random database names - weren't you doing it before with a single predetermined name?